internal/buildscripts/packaging/choco/splunk-otel-collector/tools/fluentd.ps1

```diff
@@ -0,0 +1,112 @@
+$fluentd_msi_name = "td-agent-4.1.1-x64.msi"
+$fluentd_dl_url = "https://packages.treasuredata.com/4/windows/$fluentd_msi_name"
+try {
+    Resolve-Path $env:SYSTEMDRIVE
+    $fluentd_base_dir = "${env:SYSTEMDRIVE}\opt\td-agent"
+} catch {
+    $fluentd_base_dir = "\opt\td-agent"
+}
+$fluentd_config_dir = "$fluentd_base_dir\etc\td-agent"
+$fluentd_config_path = "$fluentd_config_dir\td-agent.conf"
+$fluentd_service_name = "fluentdwinsvc"
+
+try {
+    Resolve-Path $env:TEMP
+    $tempdir = "${env:TEMP}\Fluentd"
+} catch {
+    $tempdir = "\tmp\Fluentd"
+}
+
+# download a file to a given destination
+function download_file([string]$url, [string]$outputDir, [string]$fileName) {
+    try {
+        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        (New-Object System.Net.WebClient).DownloadFile($url, "$outputDir\$fileName")
+    } catch {
+        $err = $_.Exception.Message
+        $message = "
+        An error occurred while downloading $url
+        $err
+        "
+        throw "$message"
+    }
+}
+
+# check that we're not running with a restricted execution policy
+function check_policy() {
+    $executionPolicy  = (Get-ExecutionPolicy)
+    $executionRestricted = ($executionPolicy -eq "Restricted")
+    if ($executionRestricted) {
+        throw @"
+Your execution policy is $executionPolicy, this means you will not be able import or use any scripts including modules.
+To fix this change you execution policy to something like RemoteSigned.
+        PS> Set-ExecutionPolicy RemoteSigned
+For more information execute:
+        PS> Get-Help about_execution_policies
+"@
+    }
+}
+
+# check execution policy
+Write-Host 'Checking execution policy'
+check_policy
```

Comment:
We should check the policy at the beginning of `chocolateyinstall.ps1` if  `$SkipFluentd` is false before installing anything.

Reply:
changes made.